odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -833,3 +833,107 @@ class City(models.Model):
+
+
+class SelectionExtensionDefault(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete default"
+
+    my_selection = fields.Selection(selection_add=[
+        ('baz', "Baz"),
+    ], ondelete={'baz': 'default'})
+
+
+class SelectionExtensionCascade(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete cascade"
+
+    my_selection = fields.Selection(selection_add=[
+        ('eggs', "Eggs"),
+    ], ondelete={'eggs': 'cascade'})
+
+
+class SelectionExtensionLiteral(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete set <option>"
+
+    my_selection = fields.Selection(selection_add=[
+        ('bacon', "Bacon"),
+    ], ondelete={'bacon': 'set bar'})
+
+
+class SelectionExtensionMultipleExplicit(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete set <option>"
+
+    my_selection = fields.Selection(selection_add=[
+        ('pikachu', "Pikachu"),
+        ('eevee', "Eevee"),
+    ], ondelete={'pikachu': 'default', 'eevee': 'set bar'})
+
+
+class SelectionExtensionMultipleImplicit(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete set <option>"
+
+    my_selection = fields.Selection(selection_add=[
+        ('charmander', "Charmander"),
+        ('squirtle', "Squirtle"),
+        ('bulbasaur', "Bulbasaur"),
+    ], ondelete={'charmander': 'default', 'squirtle': 'default', 'bulbasaur': 'default'})
+
+class SelectionBase(models.Model):
+    _name = 'test_new_api.model_selection_base'
+    _description = "Dummy model with a base selection field"
+
+    my_selection = fields.Selection([
+        ('foo', "Foo"),
+        ('bar', "Bar"),
+    ])
+
+
+class SelectionExtensionNullExplicit(models.Model):
+    _inherit = 'test_new_api.model_selection_base'
+    _description = "Dummy model with a selection field extension with ondelete null"
+
+    my_selection = fields.Selection(selection_add=[
+        ('quux', "Quux"),
+    ], ondelete={'quux': 'null'})
+
+
+class SelectionExtensionNullImplicit(models.Model):
+    _inherit = 'test_new_api.model_selection_base'
+    _description = "Dummy model with a selection field extension without ondelete clause"
+
+    my_selection = fields.Selection(selection_add=[
+        ('ham', "Ham"),
+    ])
+
+
+class SelectionExtensionCallback(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+
+    my_selection = fields.Selection(selection_add=[
+        ('knickers', "Oh la la"),
+    ], ondelete={'knickers': lambda recs: recs.write({'active': False, 'my_selection': 'foo'})})
```

Comment:
Maybe move this class up (with the other extensions of `model_selection_base_required`.